### PR TITLE
Anim export

### PR DESCRIPTION
**Pull request by ssbucarlos**
Going to merge soon since only thing functionally remaining is camera export and since theres alot of misc bugfixes / qol features also in here. @ScanMountGoat lmk if theres anything that should be done before merging.

### CODE REVIEW

**Comment by ScanMountGoat** [comment]
Thanks for taking the time to finish this! The overall changes look ok. I'll try and set aside some time this week to go over things more closely before we merge.

I think it would be best to change the hlpb field names in the addon UI to match SSBH Editor, ssbh_data, and the next ssbh_data_py update. You can find the unreleased Python API at the link below. These field names probably won't change anytime soon. 
https://github.com/ScanMountGoat/ssbh_data_py/blob/main/ssbh_data_py/ssbh_data_py/hlpb_data.pyi

There won't be a need to update the UI right away even if the field names change in the future. I mostly just want the existing names to be updated since they have confusing phrases inherited from years ago like "root" vs "parent". The bone effected by the constraint is the "target" bone. This makes the JSON code a little harder to read, but it will be replaced at some point by ssbh_data_py anyway.

**Comment by ssbucarlos** [comment]
Ok camera export is done along with the helper bone property name update. 
The only issue remaining is figuring out the proper blender FOV, but thats a long-standing issue that existed even with the previous plugin. I can push a fix for it later when i figure it out.

**Comment by ScanMountGoat** [comment]
The only thing that I think needs to be addressed before merging is the orientation issues on the foot when viewing a00wait2.nuanmb for Mario. The issue doesn't seem to persist when exporting and viewing in SSBH Editor. This could just be a visual thing, but it would be nice to have some idea as to what's going on.

**Comment by TabuuForteAkugun** [comment]
Woo, neat! I'm curious about something: with this NEW plugin is there a way to edit the Z-axis translation on the Trans bone? With the old tool, I have many problems with scaling the animation to fit the scale of the model (exported from Max). When I scale the animation, the animation's Trans values are also scaled. How can I stop this?

**Comment by ssbucarlos** [comment]
> The only thing that I think needs to be addressed before merging is the orientation issues on the foot when viewing a00wait2.nuanmb for Mario. The issue doesn't seem to persist when exporting and viewing in SSBH Editor. This could just be a visual thing, but it would be nice to have some idea as to what's going on.

I dont notice anything on my end when i loaded a00wait2.nuanmb for mario, do u know on which frame the issue is and on which bone it is?

**Comment by ssbucarlos** [comment]
> Woo, neat! I'm curious about something: with this NEW plugin is there a way to edit the Z-axis translation on the Trans bone? 
Yes the translation values for the root bones are now editable.

>  With the old tool, I have many problems with scaling the animation to fit the scale of the model (exported from Max). When I scale the animation, the animation's Trans values are also scaled. How can I stop this?
If you imported the animation from Max then im not sure, maybe mess around with the Max export settings or blender FBX importer settings to see if that can fix the issue.

**Comment by TabuuForteAkugun** [comment]
Cool, I was nervous you wouldn't understand what I was trying to get at (bad wording :P)

But yeah. The MODEL exported from max, and the ANIMATION was with your old edited Blender script. Not Max.

File: modules/import_anim.py [hunk 1]
```diff
@@ -126,33 +110,30 @@ def import_model_anim(context, filepath,
     frame_count = ssbh_anim_data.final_frame_index + 1
 
     scene = context.scene
+    arma = scene.sub_scene_properties.anim_import_arma
     scene.frame_start = first_blender_frame
     scene.frame_end = scene.frame_start + frame_count - 1
     scene.frame_set(scene.frame_start)
     bpy.ops.object.mode_set(mode='OBJECT', toggle=False) # whatever object is currently selected, exit whatever mode its in
```
Comment (ScanMountGoat) [inline modules/import_anim.py:117]: This line fails if nothing is active or selected.

**Comment by ScanMountGoat** [comment]
The GitHub UI is being finicky. On line 117 in import_anim.py, the mode set can fail if nothing is active or selected. The easiest way to reproduce this is to import a model and then import an anim without selecting anything in Blender.

**Comment by ssbucarlos** [comment]
> The GitHub UI is being finicky. On line 117 in import_anim.py, the mode set can fail if nothing is active or selected. The easiest way to reproduce this is to import a model and then import an anim without selecting anything in Blender.

Good catch, ill fix it rn

**Comment by ScanMountGoat** [comment]
> > The only thing that I think needs to be addressed before merging is the orientation issues on the foot when viewing a00wait2.nuanmb for Mario. The issue doesn't seem to persist when exporting and viewing in SSBH Editor. This could just be a visual thing, but it would be nice to have some idea as to what's going on.
> 
> I dont notice anything on my end when i loaded a00wait2.nuanmb for mario, do u know on which frame the issue is and on which bone it is?

Is this not happening on your end? My install may be a little weird, so I wanted to double check.
image

**Comment by ssbucarlos** [comment]
> But yeah. The MODEL exported from max, and the ANIMATION was with your old edited Blender script. Not Max.

The animation importer is intended to be used with vanilla skeletons also imported with this plugin. If the other model is imported from another format, if u run into any animation import issues when importing with smush_blender i think besides matching the imported model scale to match the vanilla ultimate model (if scale is really the only issue), ur best bet may be to use an animation retargetting plugin like rokoko.

**Comment by ssbucarlos** [comment]
> > > The only thing that I think needs to be addressed before merging is the orientation issues on the foot when viewing a00wait2.nuanmb for Mario. The issue doesn't seem to persist when exporting and viewing in SSBH Editor. This could just be a visual thing, but it would be nice to have some idea as to what's going on.
> > 
> > 
> > I dont notice anything on my end when i loaded a00wait2.nuanmb for mario, do u know on which frame the issue is and on which bone it is?
> 
> Is this not happening on your end? My install may be a little weird, so I wanted to double check. image

that definitely didnt happen on my end, but im deleting my mario files and re-extracting just to double check

**Comment by TabuuForteAkugun** [comment]
> > But yeah. The MODEL exported from max, and the ANIMATION was with your old edited Blender script. Not Max.
> 
> The animation importer is intended to be used with vanilla skeletons also imported with this plugin. If the other model is imported from another format, if u run into any animation import issues when importing with smush_blender i think besides matching the imported model scale to match the vanilla ultimate model (if scale is really the only issue), ur best bet may be to use an animation retargetting plugin like rokoko.


I gotcha. Looks like I'll have to re-export all my models with your plugin as well as my new ones. Thanks!

**Comment by ScanMountGoat** [comment]
> > > > The only thing that I think needs to be addressed before merging is the orientation issues on the foot when viewing a00wait2.nuanmb for Mario. The issue doesn't seem to persist when exporting and viewing in SSBH Editor. This could just be a visual thing, but it would be nice to have some idea as to what's going on.
> > > 
> > > 
> > > I dont notice anything on my end when i loaded a00wait2.nuanmb for mario, do u know on which frame the issue is and on which bone it is?
> > 
> > 
> > Is this not happening on your end? My install may be a little weird, so I wanted to double check. image
> 
> that definitely didnt happen on my end, but im deleting my mario files and re-extracting just to double check

I can reliably reproduce this on Blender 2.93 but not Blender 3.0.

**Comment by ssbucarlos** [comment]
> I can reliably reproduce this on Blender 2.93 but not Blender 3.0.

Now that u mention it, i used to run into random bone twist issues.

**Comment by ScanMountGoat** [comment]
Hopefully this is an easy fix. If not, we can increment the addon version and supported Blender version and document it in the readme. The update to Blender 3.1+ will happen soon anyway. I don't think many people are on 2.93 anyway. Blender doesn't make it easy to find older versions.

**Comment by ssbucarlos** [comment]
yeah i think its just a glitch on blenders side anyways thats already been fixed, i dont think we need to do anything. Especially if we do move to 3.1+ soon

**Review by ScanMountGoat — APPROVED** [review]

**Comment by ScanMountGoat** [comment]
This should be good to merge. We can fix minor stuff as it comes up before the update to 3.1+.

**Pull request merged**